src/features/warpCore/warpCoreConfig.ts

```diff
@@ -99,7 +104,27 @@ function reduceOptions(optionsList: Array<WarpCoreConfig['options']>): WarpCoreC
   }, {});
 }
 
-// Remove tokens that have no connections from the token list
+// Remove tokens that have no connections from the token list, but preserve tokens that are destinations
 function filterUnconnectedToken(tokens: WarpCoreConfig['tokens']): WarpCoreConfig['tokens'] {
-  return tokens.filter((token) => token.connections && token.connections.length);
+  const destinationTokenIds = new Set<string>();
+
+  tokens.forEach((token) => {
```

Comment:
not a big deal but can you just do 
```
if (token.connections?.length) {
```
or if you want to be fancy

```suggestion
type TokenWithConnections = WarpCoreConfig['tokens'][number] & { connections: { token: string }[] };


tokens
    .filter((token): token is TokenWithConnections => !!token.connections?.length)
    .forEach((token) => {
      token.connections.forEach((conn) => {
        destinationTokenIds.add(conn.token);
      });
    });
```